From adcf25f996f8ac6af17703e92c9aa18ad7f96163 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 01:32:37 -0400
Subject: [iommu] vt-d: print correct domain id of static identity domain

Message-id: <20140905013237.30867.27385.stgit@gir.stowe>
Patchwork-id: 91279
O-Subject: [RHEL7.1 PATCH 15/28] iommu/vt-d, trivial: print correct domain id of static identity domain
Bugzilla: 1129808
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1129808
Upstream Status: 9544c003e85f6ac6b0b617e15266fe2e81caa42a

commit 9544c003e85f6ac6b0b617e15266fe2e81caa42a
Author: Jiang Liu <jiang.liu@linux.intel.com>
Date:   Mon Jan 6 14:18:13 2014 +0800

    iommu/vt-d, trivial: print correct domain id of static identity domain

    Field si_domain->id is set by iommu_attach_domain(), so we should only
    print domain id for static identity domain after calling
    iommu_attach_domain(si_domain, iommu), otherwise it's always zero.

    Signed-off-by: Jiang Liu <jiang.liu@linux.intel.com>
    Signed-off-by: Joerg Roedel <joro@8bytes.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/intel-iommu.c b/drivers/iommu/intel-iommu.c
index b0a35e9..6ca47a3 100644
--- a/drivers/iommu/intel-iommu.c
+++ b/drivers/iommu/intel-iommu.c
@@ -2248,8 +2248,6 @@ static int __init si_domain_init(int hw)
  if (!si_domain)
   return -EFAULT;
 
- pr_debug("Identity mapping domain is domain %d\n", si_domain->id);
-
  for_each_active_iommu(iommu, drhd) {
   ret = iommu_attach_domain(si_domain, iommu);
   if (ret) {
@@ -2264,6 +2262,8 @@ static int __init si_domain_init(int hw)
  }
 
  si_domain->flags = DOMAIN_FLAG_STATIC_IDENTITY;
+ pr_debug("IOMMU: identity mapping domain is domain %d\n",
+   si_domain->id);
 
  if (hw)
   return 0;
-- 
1.7.1